### DISCUSSION TITLE
Tesla Charging Consumption

### DISCUSSION THREAD

**Discussion by joaomribeiro24** [alandtse/tesla#812] [General]
Hey there!

What should be the sensor to use (in W) to see power consumption per day?
I have a chart on Grafana with Total Consumption of my house (W) and Total Consumption of Heating System (W)
What should be the Tesla one?

Thanks

**Comment by pacemaker82** [comment]
I dont have an answer but you want kWh for energy consumption (power over time), which is what you pay for. There is the sensor.energy_added but it resets every charge session (plug in/unplug), so I am not sure how to get it from that without manually doing the maths.

**Comment by carryonrewardless** [comment]
How are you getting your Tesla data into Grafana? Are you using Home Assistant?  If so, you can use the "utility_meter" in HA to integrate the charger_power entity.  You can set this up as a daily entity that resets each day.
HTH
